wagtail/admin/tests/test_account_management.py

```diff
@@ -419,6 +419,20 @@ def test_change_password_post_password_mismatch(self):
         self.user.refresh_from_db()
         self.assertTrue(self.user.check_password("password"))
 
+    def test_ignore_change_password_if_only_old_password_supplied(self):
+        response = self.post_form(
+            {
+                "password-old_password": "password",
```

Comment:
Just a nitpick - we should also pass the "new password" fields here as empty fields, to ensure that the test fully matches what would happen in a real form:

```suggestion
                "password-old_password": "password",
                "password-new_password1": "",
                "password-new_password2": "",
```

Other than that, all looks good to me!

Reply:
Oh. I tried that initially in the unit test, and compared it with the payload from the browser (when a user actually does it). The form treats blank strings separately from no values being passed at all.
If I added those, it would be an entirely different test scenario.

Reply:
To add more context: when I did that, the password actually got changed to a blank string (because it treats them differently from not having any values passed at all.)

Reply:
Hmm, that doesn't seem right - leaving the fields out of the POST body would be equivalent to those fields not being part of the form at all. Passing them as empty strings means that these fields _are_ part of the form but the user has left them blank - and that's the scenario we want to test here, isn't it? If the test is failing with that setup, then it sounds like the fix isn't working as intended.

Reply:
Not quite. Let me explain the scenario a bit better. I'll explain the problem, the logic I used in going about the fix, and then the test after the fix.

Problem: Once a user opens their account management, some browsers auto fill the "old password" field by default, whether or not the user wants to change to a new password. So when they hit the save button, the form assumes the user wanted to change the password (because the old password field isn't empty) and triggers a validation error (because the 'new password' fields would be blank).

Fix: Since the 'password-change' panel is a smaller form on its own, I removed the 'old password' field from binding the smaller form to the larger form (the whole account management form). So, even if the old password is autofilled or a user passes something into it and then hits the 'save' button, nothing will happen, provided nothing was entered into the 'new password' field(s).

Test: I'm basically testing that when only the 'old password' field has something in it, it should be ignored. The only thing that should trigger password validation should be when there are things in the 'new password' field that actually require validation.

Do you understand me better?

Reply:
Yes, I agree with everything you wrote here. So, if the user submits a form consisting of:

* old_password: anything
* new_password1: blank
* new_password2: blank

the form should be ignored - no validation error, and the password should not change. If a user submits a form like that, then the browser will send the empty fields as part of the form submission. I just tested this with Chrome developer tools to confirm:
![Screenshot 2023-10-23 at 15 16 53](https://github.com/wagtail/wagtail/assets/85097/0702d5cc-0063-4cac-8fbe-3a306c8f8700)

Therefore, to match the scenario seen in the browser, the POST data in the test should also include the two new_password fields as empty strings, as shown in my suggestion above. I just tried this out and the test still passes, showing that your fix to account.py is indeed correct when tested against the "real world" setup.

Reply:
I just tried it out as you suggested, and it did infact work.
😅 I got in the habit of switching back to the main branch every time I made a push, so when I first wrote the tests (after the initial push with the new logic), I mistakenly ran them with the main branch, instead of the one that contained the new logic. That was why it failed at the time I tried it.
Apologies for the back and forth, and thank you so much for taking your time to walk me through it.
I've pushed your suggested change.